src/geo/transform.js

```diff
@@ -1081,6 +1081,9 @@ class Transform {
      * @private
      */
     locationPoint3D(lnglat: LngLat): Point {
+        if (this.projection.name === "globe") {
+            return this.locationPoint(lnglat);
+        }
         return this._coordinatePoint(this.locationCoordinate(lnglat), true);
```

Comment:
In [projection.js](https://github.com/mapbox/mapbox-gl-js/blob/9e604f8d5bfdf72ad94a7fb55e64109712b3d23e/src/geo/projection/projection.js#L77_L79) we have `locationPoint` resolving to:

```
locationPoint(tr: Transform, lngLat: LngLat): Point {
  return tr._coordinatePoint(tr.locationCoordinate(lngLat), false);
}
```

To me this says that by adding this extra statement:
```
if (this.projection.name === "globe") {
  return this.locationPoint(lnglat);
}
```
You're effectively switching to:
```
return this._coordinatePoint(this.locationCoordinate(lnglat), false);
```
and telling it not to sample 3d terrain. Which might make sense since the 3d terrain sampling with globe is probably not working properly when used solely for deferred rendering.

Instead of an if statement, how about either:

1. Fixing 3d terrain sampling when globe is enabled [here](https://github.com/mapbox/mapbox-gl-js/blob/9e604f8d5bfdf72ad94a7fb55e64109712b3d23e/src/geo/transform.js#L1255).

2. Wrap this function in the projection API so that you can disable 3d terrain sampling there. So here we would have `return this.projection.locationPoint3D`, then `projection.js` we would have:

```
locationPoint3D(tr: Transform, lngLat: LngLat): Point {
  return tr._coordinatePoint(tr.locationCoordinate(lngLat), true);
}
```
And in globe, override it for:
```
locationPoint3D(tr: Transform, lngLat: LngLat): Point {
  return tr._coordinatePoint(tr.locationCoordinate(lngLat), false);
}
```